### PR TITLE
create-react-app

### LINKED ISSUE #9: [FEATURE] Merge With Create-React-App Template
Merge current repo setup with the setup from create-react-app without breaking or removing any functionalities from the previous setup

### PR DESCRIPTION
**Pull request by amjedidiah**
**What the PR achieves** 
This PR does the following:

- Integrates the **create-react-app** setup with the existing **js-template** setup without breaking any of the existing testing or linting functions

- Adds a feature that checks for linting && test errors before starting, building, or deploying the app: locally or online.
It also runs this check before committing to Git.
If any lint or test errors are found, the following start, build, deploy or commit process is prevented

**Steps to test the PR locally**
1. Create a linting or test error on /src/app.js and run any of the following:

- ` npm start`
- `npm run build`
- `npm run deploy`

2. Make changes that creates an error and try to commit the change

**An npm error is thrown if an error exists inside the `/src` for the above 2 steps**

The check can be done solely by running `npm run check` 

Closes #9

### CODE REVIEW

**Review by okonwanjiphillip — DISMISSED** [review]

**Review by vkguru — DISMISSED** [review]

File: src/__tests__/App.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import { render } from '@testing-library/react';
+import App from '../App.js';
```
Comment (sam-karis) [inline src/__tests__/App.test.js]:
```suggestion
import App from '../app.js';
```

File: src/index.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
-import app from './app';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App.js';
```
Comment (sam-karis) [inline src/index.js]:
```suggestion
import App from './app.js';
```

File: public/manifest.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (sam-karis) [inline public/manifest.json]: Please remove files you are not using e.g `public/logo512.png, public/logo192.png and public/favicon.ico`

**Review by sam-karis — CHANGES_REQUESTED** [review]
Also on the `PR` description please what the PR achieves and steps to test the PR locally.

Lastly ensure before pushing the code passes `npm run lint and npm run test`

File: src/__tests__/App.test.js (hunk 1 above)
Comment (amjedidiah) [reply src/__tests__/App.test.js:4]: Changing the case of the App.js file throws an eslint error

Comment (amjedidiah) [reply src/__tests__/App.test.js]: Changing the case of the App.js file throws an eslint error.
Something about not being the convention with react, where all component names must start with an uppercase letter

File: src/index.js (hunk 2 above)
Comment (amjedidiah) [reply src/index.js]: Changing the case of the App.js file throws an eslint error.
Something about not being the convention with react, where all component names must start with an uppercase letter

File: src/__tests__/App.test.js (hunk 1 above)
Comment (sam-karis) [reply src/__tests__/App.test.js]: Then in that case rename the file under `src/app.js` to `src/App.js`

Comment (amjedidiah) [reply src/__tests__/App.test.js]: Done

**Comment by amjedidiah** [comment]
> Also on the `PR` description please what the PR achieves and steps to test the PR locally.
> 
> Lastly ensure before pushing the code passes `npm run lint and npm run test`

Done this

File: src/__tests__/App.test.js (hunk 1 above)
Comment (amjedidiah) [reply src/__tests__/App.test.js]: Finally went with the lowercase naming convention for the app file

File: src/index.js (hunk 2 above)
Comment (amjedidiah) [reply src/index.js]: Finally went with the lowercase naming convention for the app file

File: public/manifest.json (hunk 3 above)
Comment (amjedidiah) [reply public/manifest.json]: Done.
Replaced unused files with the proper files of our own

**Review by vkguru — APPROVED** [review]

**Review by sam-karis — APPROVED** [review]

**Review by okonwanjiphillip — APPROVED** [review]

**Comment by amjedidiah** [comment]
Thanks a lot @sam-karis @okonwanjiphillip @vkguru

**Review by okonwanjiphillip — APPROVED** [review]

**Pull request merged**